GpsDataCapturer/sharedlibrary/src/main/java/com/google/sharedlibrary/FusedLocationProviderHelper.java

```diff
@@ -0,0 +1,85 @@
+package com.google.sharedlibrary;
+
+import android.annotation.SuppressLint;
+import android.os.Looper;
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.location.FusedLocationProviderClient;
+import com.google.android.gms.location.LocationCallback;
+import com.google.android.gms.location.LocationRequest;
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.Task;
+
+/**
+ * Helper class that help the GpsDataCaptureService to start/stop the FusedLocationProviderClient.
+ */
+public class FusedLocationProviderHelper {
+    private static final String TAG = "FLPHelper";
+    private static final int INTERVAL = 1000;
+    private static final int FASTEST_INTERVAL = 1000;
+    private static LocationRequest locationRequest;
+
+    /**
+     * Start fusedLocationProviderClient
+     */
+    @SuppressLint("MissingPermission")
+    public static void startFusedLocationProviderClient(
+            FusedLocationProviderClient fusedLocationProviderClient,
+            LocationCallback locationCallback) {
+        createLocationRequest();
+
+        fusedLocationProviderClient.requestLocationUpdates(locationRequest, locationCallback,
+                Looper.getMainLooper())
+                .addOnFailureListener(new OnFailureListener() {
+                    @Override
+                    public void onFailure(@NonNull Exception e) {
+                        Log.e(TAG, "FusedLocationProviderClient could not be started.", e);
+                    }
+                })
+                .addOnCompleteListener(new OnCompleteListener<Void>() {
+                    @Override
+                    public void onComplete(@NonNull Task<Void> task) {
+                        Log.d(TAG,
+                                "FusedLocationProviderClient request location update completed.");
+                    }
+                });
+    }
+
+    /**
+     * Create location request
+     */
+    private static void createLocationRequest() {
```

Comment:
this should be function returning LocationRequest. Don't store output in global field.